src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectWriter.cs

```diff
@@ -984,7 +984,7 @@ private long GetAssemblyId(WriteObjectInfo objectInfo)
             {
                 assemId = 0;
             }
-            else if (assemblyString.Equals(Converter.s_urtAssemblyString))
+            else if (assemblyString.Equals(Converter.s_urtAssemblyString) || assemblyString.Equals(Converter.s_urtAlternativeAssemblyString))
```

Comment:
Had another wild idea,  What if we put TypeForwardedFrom("mscorlib") on anything that was in mscorlib in desktop?  That would address case 2, but not case 1.  I suppose for case 1, if the type existed on desktop we could do the same (eg TypeForwardedFrom("system")) and if the type didn't exist on desktop then it is "who cares".